app/lib/screens/signing/sign_with_qrcode.dart

```diff
@@ -0,0 +1,327 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+import 'package:threebotlogin/helpers/logger.dart';
+import 'package:threebotlogin/models/wallet.dart';
+import 'package:threebotlogin/providers/wallets_provider.dart';
+import 'package:threebotlogin/screens/scan_screen.dart';
+import 'package:threebotlogin/services/signing_service.dart';
+import 'package:threebotlogin/widgets/custom_dialog.dart';
+
+class SignWithQRCodeScreen extends ConsumerStatefulWidget {
+  const SignWithQRCodeScreen({super.key});
+
+  @override
+  ConsumerState<SignWithQRCodeScreen> createState() =>
+      _SignWithTextScreenState();
+}
+
+class _SignWithTextScreenState extends ConsumerState<SignWithQRCodeScreen> {
+  final TextEditingController _textController = TextEditingController();
+  Wallet? selectedWallet;
+  String? signedData;
+  bool isLoading = false;
+  String? scannedDataError;
+  String? walletError;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _textController.dispose();
+    super.dispose();
+  }
+
+  void _validateInputs() {
+    setState(() {
+      if (_textController.text.isEmpty) {
+        scannedDataError = 'Please scan a QR code first';
+      } else {
+        scannedDataError = null;
+      }
+
+      if (selectedWallet == null) {
+        walletError = 'Please select a wallet';
+      } else {
+        walletError = null;
+      }
+    });
+  }
+
+  void _showInvalidQRCodeDialog() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) => CustomDialog(
+        type: DialogType.Error,
+        image: Icons.error,
+        title: 'Invalid QR Code',
+        description:
+            'The QR code is missing the required information or invalid.',
+        actions: [
+          TextButton(
+            child: const Text('Close'),
+            onPressed: () {
+              Navigator.pop(context);
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  scanQrCode() async {
+    await SystemChannels.textInput.invokeMethod('TextInput.hide');
+    // QRCode scanner is black if we don't sleep here.
+    bool slept =
+        await Future.delayed(const Duration(milliseconds: 400), () => true);
+    late Barcode result;
+    if (slept) {
+      if (context.mounted) {
+        result = await Navigator.push(context,
+            MaterialPageRoute(builder: (context) => const ScanScreen()));
+      }
+    }
+    if (result.rawValue != null) {
+      late final Uri code;
+      try {
+        code = Uri.parse(result.rawValue!);
```

Comment:
In the QR code, it won't be a URL, it can be JSON data with 3 keys 

`content` -> direct text content to be signed
`src` -> a URL to get its content, then sign the extracted data
`dest` -> send a post request to a server URL with key `signature`